src/Workspaces/Core/Portable/Workspace/Solution/StateChecksums.cs

```diff
@@ -120,33 +120,35 @@ public static SolutionCompilationStateChecksums Deserialize(ObjectReader reader)
-        if (searchingChecksumsLeft.Remove(this.Checksum))
-            result[this.Checksum] = this;
+        if (assetPath.IncludeSolution)
+        {
+            if (searchingChecksumsLeft.Remove(this.Checksum))
+                result[this.Checksum] = this;
 
-        if (searchingChecksumsLeft.Remove(this.SourceGeneratorExecutionVersionMap))
-            result[this.SourceGeneratorExecutionVersionMap] = compilationState.SourceGeneratorExecutionVersionMap;
+            if (searchingChecksumsLeft.Remove(this.SourceGeneratorExecutionVersionMap))
+                result[this.SourceGeneratorExecutionVersionMap] = compilationState.SourceGeneratorExecutionVersionMap;
 
-        if (searchingChecksumsLeft.Count == 0)
-            return;
+            if (searchingChecksumsLeft.Count == 0)
+                return;
 
-        if (compilationState.FrozenSourceGeneratedDocumentStates != null)
-        {
-            Contract.ThrowIfFalse(FrozenSourceGeneratedDocumentIdentities.HasValue);
+            if (compilationState.FrozenSourceGeneratedDocumentStates != null)
+            {
+                Contract.ThrowIfFalse(FrozenSourceGeneratedDocumentIdentities.HasValue);
 
-            // This could either be the checksum for the text (which we'll use our regular helper for first)...
-            await ChecksumCollection.FindAsync(compilationState.FrozenSourceGeneratedDocumentStates, assetHint.DocumentId, searchingChecksumsLeft, result, cancellationToken).ConfigureAwait(false);
+                // This could either be the checksum for the text (which we'll use our regular helper for first)...
+                await ChecksumCollection.FindAsync(compilationState.FrozenSourceGeneratedDocumentStates, hintDocument: null, searchingChecksumsLeft, result, cancellationToken).ConfigureAwait(false);
 
-            // ... or one of the identities. In this case, we'll use the fact that there's a 1:1 correspondence between the
-            // two collections we hold onto.
-            for (var i = 0; i < FrozenSourceGeneratedDocumentIdentities.Value.Count; i++)
-            {
-                var identityChecksum = FrozenSourceGeneratedDocumentIdentities.Value[0];
-                if (searchingChecksumsLeft.Remove(identityChecksum))
+                // ... or one of the identities. In this case, we'll use the fact that there's a 1:1 correspondence between the
+                // two collections we hold onto.
+                for (var i = 0; i < FrozenSourceGeneratedDocumentIdentities.Value.Count; i++)
                 {
-                    var id = FrozenSourceGeneratedDocuments!.Value.Ids[i];
-                    Contract.ThrowIfFalse(compilationState.FrozenSourceGeneratedDocumentStates.TryGetState(id, out var state));
-                    result[identityChecksum] = state.Identity;
+                    var identityChecksum = FrozenSourceGeneratedDocumentIdentities.Value[0];
+                    if (searchingChecksumsLeft.Remove(identityChecksum))
```

Comment:
> if (searchingChecksumsLeft.Remove(identityChecksum)) [](http://example.com/codeflow?start=19&length=53)

Just wanting to make sure the assetPath filter check is appropriate for finding these

Reply:
yes.  these are 'solution level' assets to sync.